src/main/java/be/uantwerpen/scicraft/entity/ElectronEntity.java

```diff
@@ -46,55 +50,39 @@ protected Item getDefaultItem() {
 
     /**
      * TODO change ths snowball particle effect to custom particle effect?
+     *
      * @return ParticleEffect used on collision
      */
     @Environment(EnvType.CLIENT)
     private ParticleEffect getParticleParameters() { // Needed for particles on collision with the world
         ItemStack itemStack = this.getItem();
-        return (ParticleEffect)(itemStack.isEmpty() ? ParticleTypes.ITEM_SNOWBALL : new ItemStackParticleEffect(ParticleTypes.ITEM, itemStack));
+        return (ParticleEffect) (itemStack.isEmpty() ? ParticleTypes.ITEM_SNOWBALL : new ItemStackParticleEffect(ParticleTypes.ITEM, itemStack));
     }
 
     /**
      * Handle status 3 -> generate particles onCollision
+     *
      * @param status byte with status to handle
      */
     @Environment(EnvType.CLIENT)
     public void handleStatus(byte status) { // Needed for particles on collision with the world
         if (status == 3) {
             ParticleEffect particleEffect = this.getParticleParameters();
 
-            for(int i = 0; i < 8; ++i) {
+            for (int i = 0; i < 8; ++i) {
                 this.world.addParticle(particleEffect, this.getX(), this.getY(), this.getZ(), 0.0D, 0.0D, 0.0D);
             }
         }
     }
 
-    /**
-     * Removes the initial random adjustment on throw of projectiles in minecraft
-     * @param x distance along x-axis
-     * @param y distance along y-axis
-     * @param z distance along z-axis
-     * @param speed velocity of the item
-     * @param divergence unused parameter
-     */
-    @Override
-    public void setVelocity(double x, double y, double z, float speed, float divergence) {
-        Vec3d vec3d = (new Vec3d(x, y, z)).normalize().multiply((double)speed);
-        this.setVelocity(vec3d);
-        double d = vec3d.horizontalLength();
-        this.setYaw((float)(MathHelper.atan2(vec3d.x, vec3d.z) * 57.2957763671875D));
-        this.setPitch((float)(MathHelper.atan2(vec3d.y, d) * 57.2957763671875D));
-        this.prevYaw = this.getYaw();
-        this.prevPitch = this.getPitch();
-    }
-
```

Comment:
I don't fully understand why this method has to be removed. If you take a look at the setVelocity method in ProjectileEntity class the divergence is applied and that is what we don't want? The divergence in ProjectileEntity class adds a small random offset to the start velocity of the throw. By overriden that method I remove that small random offset to the start velocity.

Reply:
Yes this function supports adding divergence, but I don't think it is called anywhere with a non-zero value for divergence for the ElectronEntity.

If i'm not mistaken, you already took care of this here: https://github.com/ScicraftLearn/Scicraft/blob/f08ed04d0c0c84510462727fb20ccc1e566505db/src/main/java/be/uantwerpen/scicraft/item/ElectronItem.java#L40

Let me know if I missed anything. I find it hard to experimentally verify whether the electron flies straight.

Reply:
It is not the case for dispenser at the moment, but I see you took care by opening another issue for that. If we want it for dispenser it is indeed better to keep the original setVelocity method and override getVariation in dispenserBehavior to get a zero divergence.